services/datamanager/datasync/sync.go

```diff
@@ -28,6 +28,8 @@ var (
 	InitialWaitTimeMillis = atomic.NewInt32(1000)
 	// RetryExponentialFactor defines the factor by which the retry wait time increases.
 	RetryExponentialFactor = atomic.NewInt32(2)
+	// OfflineWaitTimeSeconds defines the amount of time to wait to retry if the machine is offline
+	OfflineWaitTimeSeconds = atomic.NewInt32(30)
```

Comment:
Perhaps one minute? I wonder if we should have a maximum # of times that we'll retry offline before backing off, e.g. only do so for up to an hour — but it's incredibly variable on the use case, so we can always add that later if needed.

Could you manually test having many collectors/files offline and ensure that it doesn't brick the system? You can add thousands of dummy files into a collector folder and go offline to test.

Reply:
Sure! We can play around with this value and there's lots of room to figure out how best to handle offline devices. 

Just an idea for the future: do we want to make this configurable on the UI so that we can allow different orgs to handle offline devices differently? Seems like it might not be a one-size-fits-all type problem.

I will test now!